monkey/monkey_island/cc/resources/monkey.py

```diff
@@ -87,20 +87,18 @@ def post(self, **kw):
         parent = monkey_json.get('parent')
         parent_to_add = (monkey_json.get('guid'), None)  # default values in case of manual run
         if parent and parent != monkey_json.get('guid'):  # current parent is known
-            exploit_telem = [x for x in
-                             mongo.db.telemetry.find({'telem_category': {'$eq': 'exploit'},
+            exploit_telem = list(mongo.db.telemetry.find({'telem_category': {'$eq': 'exploit'},
                                                       'data.result': {'$eq': True},
                                                       'data.machine.ip_addr': {'$in': monkey_json['ip_addresses']},
-                                                      'monkey_guid': {'$eq': parent}})]
+                                                      'monkey_guid': {'$eq': parent}}))
```

Comment:
Why is this better if we're straight converting to a list?

Reply:
It is unnecessary to use a comprehension just to loop over the iterable and create a `list/set/dict` out of it. Python has a specialized set of tools for this task: the `list/set/dict `constructors, which are faster and more readable.
The same reason goes for the next two reviews.

Reply:
Build fails since this is under-indented.